File: backend/ShiftMate.API/Models/EmployeeDayPreference.cs
namespace ShiftMate.API.Models;

public class EmployeeDayPreference
{
    public int Id { get; set; }

    public int EmployeeId { get; set; }

    public DayOfWeek DayOfWeek { get; set; }

    public bool IsPreferred { get; set; }

    public bool IsUnavailable { get; set; }

    public Employee Employee { get; set; } = null!;
}

File: frontend/src/types/employeeDayPreference.ts
export interface EmployeeDayPreference {
    id: number
    employeeId: number
    employeeName: string
    dayOfWeek: number
    isPreferred: boolean
    isUnavailable: boolean
}

export interface CreateEmployeeDayPreferenceRequest {
    employeeId: number
    dayOfWeek: number
    isPreferred: boolean
    isUnavailable: boolean
}